pkgs/development/libraries/mesa/default.nix

```diff
@@ -185,10 +188,22 @@ stdenv.mkDerivation {
     sed "/^libdir=/s,$out,$osmesa," -i $osmesa/lib/libOSMesa*.la
 
     # set the default search path for DRI drivers; used e.g. by X server
-    substituteInPlace "$dev/lib/pkgconfig/dri.pc" --replace '$(drivers)' "${driverLink}"
-  '' + optionalString (vulkanDrivers != []) ''
-    # move share/vulkan/icd.d/
+    substituteInPlace "$dev/lib/pkgconfig/dri.pc" --replace '$(drivers)' "${libglvnd.driverLink}"
+
+    # remove GLES libraries; they are provided by libglvnd
```

Comment:
You may do this instead https://github.com/NixOS/nixpkgs/pull/34282/files#diff-481f6f0a7a36656c0dc31b9ccd2202afL100

Reply:
Would that still build GLES support itself?

Reply:
Anyway, libglvnd's GLES is dispatcher only, it need real GLES somewhere

Reply:
I was baffled by this too but `glxinfo` [shows working OpenGL ES](https://github.com/NixOS/nixpkgs/pull/37369#issuecomment-374176822). My guess is that `--enable-gles*` builds actual GLES into `lib{GLX,EGL}_mesa` and Mesa's `libGLES` are wrappers somewhat similar to libglvnd's. This isn't backed by anything though. I used Arch Linux package for ~stealing ideas~inspiration and they also use `--enable-gles` and manually install only needed libraries later.